midlet.js

```diff
@@ -13,3 +13,9 @@ var MIDlet = {
 
 var config = {
 };
+
+// The base directory of the app, relative to the current page.  Normally this
+// is the directory from which the page was loaded, but some test pages load
+// from a subdirectory, like tests/fs/, and they set this accordingly such that
+// code loads files, like libs/fs-init.js, can load them from the right place.
+var APP_BASE_DIR = "./";
```

Comment:
This isn't midlet-specific, so I'd rather put it elsewhere. But I don't yet have a good place for it to go, so this is alright for now.